static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx

```diff
@@ -398,10 +399,10 @@ class WidgetQueries extends React.Component<Props, State> {
           [DisplayType.AREA, DisplayType.BAR, DisplayType.LINE].includes(displayType) &&
           query.columns?.length !== 0
         ) {
-          requestData.topEvents = TOP_N;
+          requestData.topEvents = widget.limit ?? TOP_N;
           // Aggregates need to be in fields as well
           requestData.field = [...query.columns, ...query.aggregates];
-          requestData.orderby = query.aggregates?.[0];
+          requestData.orderby = query.orderby;
```

Comment:
We can delete this because it's actually part of the body on line 391. I only introduced `query.aggregates?.[0]` because on the widget builder there wasn't a way to set this field yet so I was arbitrarily choosing the first aggregate
```suggestion
```